tests/e2e/account/pw_register_test.py

```diff
@@ -0,0 +1,215 @@
+"""Test account registration."""
+# pylint: disable=redefined-outer-name
+
+import re
+
+import pytest
+from playwright.sync_api import Page, expect
+from pytest_bdd import given, parsers, scenarios, then, when
+
+scenarios('account/register.feature')
+pytestmark = pytest.mark.usefixtures('live_server')
+
+TWOFLOWER_EMAIL = 'twoflower@example.org'
+TWOFLOWER_PHONE = '+12015550123'
+TWOFLOWER_PASSWORD = 'te@pwd3289'  # nosec
+
+pytestmark = pytest.mark.filterwarnings(
+    "ignore:Object of type <AccountPhone> not in session",
+    "ignore:Object of type <AccountEmail> not in session",
+)
+
+
+@pytest.fixture()
+def published_project(db_session, new_project):
+    new_project.publish()
+    db_session.add(new_project)
+    db_session.commit()
+    return new_project
+
+
+@pytest.fixture()
+def user_twoflower_with_password_and_contact(db_session, user_twoflower):
+    user_twoflower.password = TWOFLOWER_PASSWORD
+    user_twoflower.add_phone(TWOFLOWER_PHONE)
+    user_twoflower.add_email(TWOFLOWER_EMAIL)
+    db_session.add(user_twoflower)
+    db_session.commit()
+    return user_twoflower
+
+
+def wait_until_recaptcha_loaded(page: Page) -> None:
+    page.wait_for_selector(
+        '#form-passwordlogin > div.g-recaptcha > div > div.grecaptcha-logo > iframe',
+        timeout=10000,
+    )
+
+
+@given("Anonymous visitor is on the home page")
+def given_anonuser_home_page(live_server, page: Page) -> None:
+    page.goto(live_server.url)
+    expect(page).to_have_title("Test Hasgeek")
+
+
+@when(
+    parsers.re(
+        "they navigate to the login page and submit"
+        " (?P<phone_or_email>a phone number|an email address)"
+    ),
+    target_fixture='anon_username',
+)
+def when_anonuser_navigates_login_and_submits(
+    app, live_server, phone_or_email: str, page: Page
+) -> dict[str, str]:
+    if phone_or_email == "a phone number":
+        username = '8123456789'
+    elif phone_or_email == "an email address":
+        username = 'anon@example.com'
```

Comment:
I think you should put both these constants at the top of the file. We'll have to audit for where we're using these values across teats and put them into shared fixtures – apart from ensuring we're not using real data.